homeassistant/components/fronius/binary_sensor.py

```diff
@@ -0,0 +1,102 @@
+"""Binary sensors for Fronius devices."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.const import EntityCategory, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import FroniusEntityDescription, _FroniusEntity
+
+if TYPE_CHECKING:
+    from . import FroniusConfigEntry
+    from .coordinator import FroniusPowerFlowUpdateCoordinator
+
+
+@dataclass(frozen=True)
+class FroniusBinarySensorEntityDescription(
+    FroniusEntityDescription, BinarySensorEntityDescription
+):
+    """Describes Fronius binary_sensor entity."""
+
+    default_value: bool | None = None
+    response_key: str | None = None
```

Comment:
`response_key` is already included in FroniusEntityDescription and `default_value` isn't used in any description, so could also be removed here.